### PR TITLE
Show flying leaf animation for Walking Leaf and Leaf Shot

### PR DESCRIPTION
**Pull request by shawnanastasio**
The flying leaf animation is now shown when the Walking Leaf and Leaf Shot
are squished. This animation was previously only shown for the Poison Ivy.
In order to limit code duplication, the spawn_explosion_sprites method was
created in the MovingSprite class. For the Walking Leaf, a new set of leaf
particles was created, based off of those for the Poison Ivy, but with
adjusted color. This fixes issue #644.

### CODE REVIEW

**Comment by maxteufel** [comment]
Do we need extra particle sprite files? Shouldn't these rather be animations in the object-specific sprites?

**Comment by shawnanastasio** [comment]
@maxteufel they probably could be, but for consistency's sake I was just copying how it was done with the Poison Ivy and made new particle sprite files.

**Comment by maxteufel** [comment]
@shawnanastasio Indeed. I think the proper way this should be handled is via sprite-specific animations though. Feel free to change everything to that. That way, it's customizable per-sprite, not per-badguy, which is good.

File: src/object/moving_sprite.cpp [hunk 1]
```diff
@@ -175,4 +178,22 @@ void MovingSprite::after_editor_set()
   sprite->set_action(current_action);
 }
 
+void MovingSprite::spawn_explosion_sprites(int count, std::string sprite_path)
```
[the remark this answers is not in the document]
Comment (tobbi) [reply src/object/moving_sprite.cpp]: (If you still intend to work on this, that is)

[the remark this answers is not in the document]
Comment (shawnanastasio) [reply src/object/moving_sprite.cpp]: Sorry, I had completely forgotten about this PR. I'll go ahead and push these changes.

**Comment by tobbi** [comment]
Thank you for your contribution!

**Pull request merged**